runner_test.go

```diff
@@ -244,3 +242,72 @@ func TestRunnerIterateE2E(t *testing.T) {
 	t.Logf("Dumping stats...")
 	json.NewEncoder(os.Stdout).Encode(runner.GetStats())
 }
+
+func TestRunnerIterateE2EWithRebuildSchedule(t *testing.T) {
+
+	if os.Getenv("E2E") == "" {
+		t.Skip("e2e tests not enabled")
+		return
+	}
+	conn := mustSetupTest(t)
+
+	runner := NewRunner(
+		conn,
+		[]View{
+			&TimeseriesView{
+				Name:            "test_view_3",
+				SourceTableSpec: "test_data_1",
+				UpdateInterval:  time.Millisecond,
+				Columns: []*StandardViewColumn{
+					{CreateSpec: "ts TIMESTAMP", SelectSpec: "date_trunc('hour', ts)", IsKey: true},
+					{CreateSpec: "key1 TEXT", SelectSpec: "key1", IsKey: true},
+					{CreateSpec: "key2 TEXT", SelectSpec: "key2", IsKey: true},
+					//omit 1 key col
+					{CreateSpec: "val1 INTEGER", SelectSpec: "SUM(val1)", IsKey: false},
+					{CreateSpec: "val2 DECIMAL", SelectSpec: "SUM(val2)", IsKey: false},
+					//add one value col
+					{CreateSpec: "val3 DECIMAL", SelectSpec: "SUM(val1::DECIMAL * val2)", IsKey: false},
+				},
+				TrackBy: &DateTracker{SourceDateColumn: "ts", DestinationDateColumn: "ts", Backtrack: time.Hour},
+				Indexes: []string{
+					"ts, key1, key2",
+				},
+			},
+		},
+	)
+
+	runner.SetBackoffEnabled(time.Minute, time.Hour)
+	runner.SetScheduledRebuildEnabled(NewRebuildSchedule(time.Now().Hour(), time.Now().Minute()))
+
+	if err := runner.Setup(); err != nil {
+		t.Errorf("Runner setup failed: %s", err)
+		return
+	}
+
```

Comment:
For testing we could set a count stat for the number of rebuilds and check that is incremented after running runOnce.